src/omero_zarr/masks.py

```diff
@@ -23,53 +39,199 @@ def image_masks_to_zarr(image, args):
+
+    :return: Label image with size `mask_shape`
+
+    TODO: Move to https://github.com/ome/omero-rois/
+    """
+
+    size_t, size_c, size_z, size_y, size_x = mask_shape
+    ignored_dimensions = ignored_dimensions or set()
+    mask_shape = tuple(mask_shape)
+
+    if not labels:
+        # TODO: Set np.int size based on number of labels
+        labels = np.zeros(mask_shape, np.int64)
+
+    for d in "TCZYX":
+        if d in ignored_dimensions:
+            assert (
+                labels.shape[DIMENSION_ORDER[d]] == 1
+            ), "Ignored dimension {} should be size 1".format(d)
+        assert (
+            labels.shape == mask_shape
+        ), "Invalid label shape: {}, expected {}".format(
+            labels.shape, mask_shape
+        )
 
-    labels = np.zeros((size_t, size_z, size_y, size_x))
-    for count, shapes in enumerate(masks.values()):
+    for count, shapes in enumerate(masks):
         # All shapes same color for each ROI
+        print(count)
         for mask in shapes:
-            t = unwrap(mask.theT) or 0
-            z = unwrap(mask.theZ) or 0
-            x = int(mask.x.val)
-            y = int(mask.y.val)
-            w = int(mask.width.val)
-            h = int(mask.height.val)
-            mask_packed = mask.getBytes()
-            # convert bytearray into something we can use
-            intarray = np.fromstring(mask_packed, dtype=np.uint8)
-            binarray = np.unpackbits(intarray)
-            # truncate and reshape
-            binarray = np.reshape(binarray[: (w * h)], (h, w))
-            # ADD to the array, so zeros in our binarray don't wipe out
-            # previous masks
-            labels[t, z, y : (y + h), x : (x + w)] += binarray * (count)
+            binim_yx, (t, c, z, y, x, h, w) = _mask_to_binim_yx(mask)
+            for i_t in _get_indices(ignored_dimensions, "T", t, size_t):
+                for i_c in _get_indices(ignored_dimensions, "C", c, size_c):
+                    for i_z in _get_indices(
+                        ignored_dimensions, "Z", z, size_z
+                    ):
+                        if check_overlaps and np.any(
+                            np.logical_and(
+                                labels[
+                                    i_t, i_c, i_z, y : (y + h), x : (x + w)
+                                ].astype(np.bool),
+                                binim_yx,
+                            )
+                        ):
+                            raise Exception(
+                                "Mask {} overlaps with existing labels".format(
+                                    count
+                                )
+                            )
+                        # ADD to the array, so zeros in our binarray don't
+                        # wipe out previous masks
+                        labels[i_t, i_c, i_z, y : (y + h), x : (x + w)] += (
+                            binim_yx * count
```

Comment:
I guess on the first mask, `count` is zero.
So when we only write 1 mask at at time 'split-masks`, they are all zero.

Reply:
Good catch! So `(count+1)`, no?